jsonrpc/specification.md

```diff
@@ -0,0 +1,49 @@
+# Portal JSON-RPC specification
+
+The `*` is meant to be a placeholder to substitute different Portal sub-networks `history`, `state`, `beacon`, etc
+
+## portal_*RoutingTableInfo
+
+## portal_*AddEnr
+- Adds the Enr to the respective sub-networks routing table
+
+## portal_*GetEnr
+
+## portal_*DeleteEnr
+
+## portal_*LookupEnr
+
+## portal_*Ping
+
+## portal_*FindNodes
+
+## portal_*FindContent
+- must validate content
+
+## portal_*Offer
+- must not validate content
+- shouldn't store data on offering node
+
+## portal_*RecursiveFindNodes
+
+## portal_*GetContent
+- must validate content
+- must check local storage first before doing recursive find content
+- if recursive find content is done must do poke mechanism
+- if recursive find content is done and content meets storage criteria store content
+  
+## portal_*TraceGetContent
+- must validate content
+- must check local storage first before doing recursive find content
+- if recursive find content is done must do poke mechanism
+- if recursive find content is done and content meets storage criteria store content
+
+## portal_*Store
+- must not validate content
```

Comment:
I'm not sure, but perhaps you could still make it validate the type encoding?

I can see however that for this call you would want to avoid additional network requests.

Reply:
> I'm not sure, but perhaps you could still make it validate the type encoding?


Of course the client should validate correct type encodings I will make this more clear 

> I can see however that for this call you would want to avoid additional network requests.

This call should make 0 network requests

Reply:
For the state network, the only part of the validation that needs a network request is looking up the block header to get the state root. If we skip the state root validation we can still perform all other validations such as decoding, checking the nodeHash matches node in proof, checking structure of the proofs etc and we now do this in Fluffy.

I agree that we shouldn't make network requests but I would suggest applying as many validations as possible to reduce the risk of storing bad data in the db, anything that can be done without a network request.